docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,64 @@
+- Add annotations support. Annotations allow users to add contextual information to their dashboards and it's possible to use queries to define them. See how you can do it [here](/create-a-plugin/extend-a-plugin/enable-for-annotations).
+- When building frontend components, make sure to use [Grafana components](https://developers.grafana.com/ui/latest/index.html?path=/docs/docs-overview-intro--page) as the base and follow the [Saga Design System](https://grafana.com/blog/2023/11/07/saga-design-system-shaping-the-future-of-user-experiences-at-grafana-labs/).
+- Add query editor help. Query editors can be complex and it's useful to provide help to users. See how you can do it [here](/create-a-plugin/extend-a-plugin/add-query-editor-help).
+- Skip hidden or empty queries. This avoids executing unnecessary or wrong queries. See this [example](https://github.com/grafana/grafana/blob/fd5f66083c91b9759ae7772f99b80c9342b93290/public/app/plugins/datasource/loki/datasource.ts#L1085).
+- Specify a default query. This can be useful for users to discover how queries are written for the plugin. See an [example](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b/examples/datasource-streaming-backend-websocket/streaming-backend-websocket-plugin/src/datasource.ts#L39-L41).
+- Avoid using `console.log` in your plugin.
+
+### Frontend (only) plugins
+
+- Frontend-only data sources typically use the [Grafana proxy](/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#add-a-proxy-route-to-your-plugin) to access an external service. While this is a simple way of adding support for queries in your plugin (and doesn't require Golang knowledge), it's generally recommended to write a backend part for your plugin. The use cases can be found [here](/introduction/backend-plugins#use-cases-for-implementing-a-backend-plugin).
+
+### Backend plugins
+
+- Add support for alerts. Backend plugins have inherent support for alerts but it needs to be enabled. Simply add `"enabled": true` to your plugin.json file.
+- Use the `CallResourceHandler` interface to serve custom HTTP requests. More information in [Resource handlers](/introduction/backend-plugins#resource-handlers). This is useful, for example, when providing query builders. See an example [here](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b23645251997088ac7a1707a72d3fd9248/examples/app-with-backend/pkg/plugin/app.go#L35).
+- Add metrics to your data-source. More information [here](/introduction/backend-plugins#collect-metrics).
```

Comment:
we'll need to remember to update these when Marcus' PR merges

Reply:
which PR?

Reply:
https://github.com/grafana/plugin-tools/pull/547

Reply:
got it, the one in the next line is the one that will break.

Reply:
seems like that PR is still undergoing, we can update this doc afterwards.